meteor-slingshot/services/cloudflare-r2.js

```diff
@@ -0,0 +1,179 @@
+const crypto = Npm.require('crypto');
+const url = Npm.require('url');
+
+Slingshot.CloudflareR2 = {
+  accessId: 'AccessKeyId',
+  secretKey: 'SecretAccessKey',
+
+  directiveMatch: {
+    bucket: String,
+    accountId: String,
+    endpoint: String,
+    cdn: String,
+    region: Match.Optional(String),
+    AccessKeyId: String,
+    SecretAccessKey: String,
+    key: Match.OneOf(String, Function),
+    expire: Match.Where(function (expire) {
+      check(expire, Number);
+      return expire > 0;
+    }),
+  },
+
+  directiveDefault: {
+    region: 'auto',
+    expire: 5 * 60 * 1000,
+  },
+
+  getContentDisposition: async function (method, directive, file, meta) {
+    var getContentDisposition = directive.contentDisposition;
+
+    if (!_.isFunction(getContentDisposition)) {
+      getContentDisposition = function () {
+        var filename = file.name && encodeURIComponent(file.name);
+
+        return (
+          directive.contentDisposition ||
+          (filename &&
+            'inline; filename="' +
+              filename +
+              "\"; filename*=utf-8''" +
+              filename)
+        );
+      };
+    }
+
+    return getContentDisposition.call(method, file, meta);
+  },
+
+  upload: async function (method, directive, file, meta) {
+    try {
+      const policy = new Slingshot.StoragePolicy()
+        .expireIn(directive.expire)
+        .contentLength(0, Math.min(file.size, directive.maxSize || Infinity));
+
+      let key;
+      if (_.isFunction(directive.key)) {
+        key = await Promise.resolve(directive.key.call(method, file, meta));
+      } else {
+        key = directive.key;
+      }
+
+      const signedUrl = await this.getSignedUrl(directive, key, file.type);
+
+      const downloadUrl = `${directive.cdn}/${key}`
+
+      return {
+        upload: signedUrl,
+        download: downloadUrl,
+        postData: [],
+      };
+    } catch (error) {
+      console.error('Error in CloudflareR2 upload:', error);
+      throw error;
```

Comment:
I don't think we need the `try/catch` here if we are throwing again, if it's just for logging purposes.

Reply:
fixed